kitsune/wiki/facets.py

```diff
@@ -122,32 +122,54 @@ def _documents_for(user, locale, topics=None, products=None):
         # NOTE: It's important to use "created__range" rather than "created__gt"
         #       with Postgres, otherwise it won't use the index on the "created"
         #       field, and the "HelpfulVote" query will be massively slower.
+        # Extended votes_query
         votes_query = (
-            HelpfulVote.objects.filter(
-                revision_id__in=qs.values_list("current_revision_id", flat=True),
-                created__range=(datetime.now() - timedelta(days=30), Now()),
-                helpful=True,
-            )
+            HelpfulVote.objects.filter(revision__document__is_archived=False)
```

Comment:
I don't think this `.filter(revision__document__is_archived=False)` is necessary, given that we're already limiting the revisions to those within the `qs` and the `qs` is already filtering out the revisions of archived documents. It may even cause the `votes_query` to be a bit more costly since now it has to join to both the `Revision` and `Document` tables, so I would recommend removing this.